block-node/messaging/src/main/java/org/hiero/block/server/messaging/impl/MessagingServiceImpl.java

```diff
@@ -0,0 +1,300 @@
+                ProducerType.SINGLE,
+                new SleepingWaitStrategy());
+    }
+
+    /**
+     * Get the configuration for the messaging service. It uses the ConfigurationBuilderFactory to load the
+     * configuration from the classpath. Public as it is used in tests, but is not exported from the module.
+     *
+     * @return the configuration for the messaging service
+     * @throws IllegalStateException if no ConfigurationBuilderFactory implementation is found
+     */
+    public static MessagingConfig getConfig() {
+        ConfigurationBuilderFactory configurationBuilderFactory = null;
+        final ServiceLoader<ConfigurationBuilderFactory> serviceLoader =
+                ServiceLoader.load(ConfigurationBuilderFactory.class, MessagingServiceImpl.class.getClassLoader());
+        final Iterator<ConfigurationBuilderFactory> iterator = serviceLoader.iterator();
+        if (iterator.hasNext()) {
+            configurationBuilderFactory = iterator.next();
+        }
+        if (configurationBuilderFactory == null) {
+            throw new IllegalStateException("No ConfigurationBuilderFactory implementation found!");
+        }
+        // Load the configuration from the classpath
+        return configurationBuilderFactory
+                .create()
+                .autoDiscoverExtensions()
+                .withConfigDataType(MessagingConfig.class)
+                .build()
+                .getConfigData(MessagingConfig.class);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void sendBlockItems(final List<BlockItemUnparsed> items) {
+        blockItemDisruptor.getRingBuffer().publishEvent((event, sequence) -> event.set(items));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void registerBlockItemHandler(BlockItemHandler handler) throws IllegalStateException {
+        if (blockItemDisruptor.hasStarted()) {
+            throw new IllegalStateException("Cannot register block item handler after the service is started");
+        }
+        blockItemHandlers.add((event, sequence, endOfBatch) -> handler.handleBlockItemsReceived(event.get()));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void registerDynamicNoBackpressureBlockItemHandler(NoBackPressureBlockItemHandler handler) {
+        final RingBuffer<BlockItemBatchRingEvent> ringBuffer = blockItemDisruptor.getRingBuffer();
+        final SequenceBarrier barrier = ringBuffer.newBarrier();
+        final EventHandler<BlockItemBatchRingEvent> eventHandler = (event, sequence, endOfBatch) -> {
+            // send on the event block items
+            handler.handleBlockItemsReceived(event.get());
+            // check if the event processor is too far behind
+            final double percentageBehindHead =
+                    (100d * ((double) (barrier.getCursor() - sequence) / (double) ringBuffer.getBufferSize()));
+            if (percentageBehindHead > 80) {
```

Comment:
nit: 80% should be a configureable number an operator can set

Reply:
See above comment on why less configuration. Again in this case 80% should be a fine number if it is not we can change the code. It should not need to be adjusted often or be that sensitive. Ideally it should work fine if it was 90% or 30%. If it needs adjustment by a ops person then we have done something wrong in the design.